exercises/practice/dnd-character/dnd_character_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'minitest/autorun'
+require_relative 'dnd_character'
+
+module Helpers
+  BASE_HITPOINTS = 10
+  MODIFIER_TO_CONSTITUTION = {
+    -4 => [3],
+    -3 => [4, 5],
+    -2 => [6, 7],
+    -1 => [8, 9],
+    0 => [10, 11],
+    1 => [12, 13],
+    2 => [14, 15],
+    3 => [16, 17],
+    4 => [18]
+  }.freeze
+
+  def attributes
+    %i[strength dexterity constitution intelligence wisdom charisma]
+  end
+end
+
+class DndCharacterTest < Minitest::Test
+  include Helpers
+
+  def test_hitpoints_correctly_calculated_from_constitution_modifier
+    character = DndCharacter.new
+    constitution_modifier = character.hitpoints - BASE_HITPOINTS
+
+    assert_includes MODIFIER_TO_CONSTITUTION[constitution_modifier], character.constitution
+  end
+
+  def test_random_character_attributes
+    character = DndCharacter.new
+    allowed_range = 3..18
+    expected_hitpoints = BASE_HITPOINTS + (character.constitution / 2 - 5)
+
+    attributes.each do |attribute|
+      assert_includes allowed_range, character.send(attribute)
+    end
+
+    assert_equal expected_hitpoints, character.hitpoints
+  end
+
+  def test_attributes_being_accessed_do_not_change
+    character = DndCharacter.new
+
+    (attributes << :hitpoints).each do |attribute|
+      assert_equal character.send(attribute), character.send(attribute)
+    end
```

Comment:
How many times should we test this to have some assurance that they are stable/unchanging?

Reply:
As above, thank you for addressing this in the RFC!